inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp

```diff
@@ -828,559 +828,583 @@ void MKLDNNNormalizeL2Node::createPrimitive() {
+            normalize_kernel.reset(
+                    new jit_uni_normalize_kernel_f32<cpu::x64::avx512_common>(attrs.jcp, *attrs.kernel_attrs.get()));
         } else if (mayiuse(cpu::x64::avx2)) {
-            normalize_modulo_kernel.reset(new jit_uni_normalize_modulo_kernel_f32<cpu::x64::avx2>(jcp));
-            normalize_kernel.reset(new jit_uni_normalize_kernel_f32<cpu::x64::avx2>(jcp, *attr.get()));
+            attrs.blk_size = 8;
+            normalize_modulo_kernel.reset(new jit_uni_normalize_modulo_kernel_f32<cpu::x64::avx2>(attrs.jcp));
+            normalize_kernel.reset(
+                    new jit_uni_normalize_kernel_f32<cpu::x64::avx2>(attrs.jcp, *attrs.kernel_attrs.get()));
         } else if (mayiuse(cpu::x64::sse41)) {
-            normalize_modulo_kernel.reset(new jit_uni_normalize_modulo_kernel_f32<cpu::x64::sse41>(jcp));
-            normalize_kernel.reset(new jit_uni_normalize_kernel_f32<cpu::x64::sse41>(jcp, *attr.get()));
+            attrs.blk_size = 4;
+            normalize_modulo_kernel.reset(new jit_uni_normalize_modulo_kernel_f32<cpu::x64::sse41>(attrs.jcp));
+            normalize_kernel.reset(
+                    new jit_uni_normalize_kernel_f32<cpu::x64::sse41>(attrs.jcp, *attrs.kernel_attrs.get()));
         }
+
         if (normalize_kernel)
             normalize_kernel->create_ker();
 
         if (normalize_modulo_kernel)
             normalize_modulo_kernel->create_ker();
+    }
 
-        const auto &p = (*attr.get()).post_ops_;
-        for (int i = 0; i < p.len(); i++) {
-            auto &post_op = p.entry_[i];
-            if (post_op.is_eltwise()) {
-                eltwise_injectors_ref.push_back(std::make_shared<cpu::ref_eltwise_scalar_fwd_t>(
-                    post_op.eltwise.alg, post_op.eltwise.alpha, post_op.eltwise.beta, post_op.eltwise.scale));
-            } else if (post_op.is_depthwise()) {
-                depthwise_injectors_ref.push_back(std::make_shared<cpu::ref_depthwise_scalar_fwd_t>(
-                        post_op.depthwise.alg));
-            }
+    void exec(const uint8_t *src_ptr, uint8_t *dst_ptr) override {
+        if (attrs.jcp.is_nchw) {
+            normalize_nchw(reinterpret_cast<const in_data_t*>(src_ptr), reinterpret_cast<out_data_t*>(dst_ptr));
+        } else if (attrs.jcp.is_nhwc) {
+            normalize_nhwc(reinterpret_cast<const in_data_t*>(src_ptr), reinterpret_cast<out_data_t*>(dst_ptr));
+        } else if (attrs.jcp.is_blk) {
+            normalize_blk(reinterpret_cast<const in_data_t*>(src_ptr), reinterpret_cast<out_data_t*>(dst_ptr));
         }
     }
-}
-
-namespace {
 
-struct NormalizeContext {
-    MKLDNNNormalizeL2Node &node;
-    const uint8_t *src;
-    uint8_t *dst;
-    const SizeVector& dims;
-};
-
-}   // namespace
+private:
+    inline float epsApply(const float &modulo) const {
+        return attrs.epsMode == NormEpsMode::ADD ? modulo + attrs.eps : std::max(modulo, attrs.eps);
+    }
```

Comment:
maybe better to extract it to base class (to avoid code duplication) and pass `epsMode` and `eps` as argument